### ISSUE TITLE
Freezing issue due to V8

### ISSUE THREAD

**Issue by andrea-s** [Microsoft/ClearScript#33]
Hey,

just wanted to let you know that there is a deadlock issue that affects V8 (and ClearScript) - as detailed here: https://bugs.chromium.org/p/v8/issues/detail?id=6933

I've been noticing that sometimes CPU would get stuck to 100% (per core) since the upgrade to 5.5. After some investigation I've concluded that this is the likely culprit. Now I've been running the patched version of V8 since a couple of days and it looks like the issue is resolved.

This is what I did:
- cherry-picked commit ac0fe8ec from V8 upstream (https://github.com/v8/v8/commit/ac0fe8ec8a13493aba5fd7fe55220d23cee62562)
- recompiled V8

Basically the same thing node did here: https://github.com/nodejs/node/pull/17695

I'm attaching the corresponding patch here - has anyone noticed the same behavior with ClearScript 5.5?

[0001-interpreter-Remove-TryInstallOptimizedCode.patch,txt](https://github.com/Microsoft/ClearScript/files/1573071/0001-interpreter-Remove-TryInstallOptimizedCode.patch.txt)

**Comment by ClearScriptLib** [comment]
@andrea-s Thank you very much for that information!

We're close to releasing Version 5.5.1 paired with V8 6.3. It looks like the V8 fix you linked will not be merged into the 6.3 stable branch, so we'll modify V8Update to apply it explicitly.

Thanks again!

**Comment by ClearScriptLib** [comment]
[Version 5.5.1](https://github.com/Microsoft/ClearScript/releases/tag/5.5.1) released.
